### ISSUE TITLE
Bug when clearing axes created with twinx

### ISSUE THREAD

**Issue by astrofrog** [matplotlib/matplotlib#4160]
### Buggy case

The following example produces an empty figure:

```
import matplotlib.pyplot as plt

fig = plt.figure()
ax1 = fig.add_subplot(1,1,1)
ax2 = ax1.twinx()
ax2.cla()
ax1.plot([1,2,3])
fig.savefig('test.png')
```

![test](https://cloud.githubusercontent.com/assets/314716/6369582/1084aecc-bce8-11e4-970a-a7e7750b8c74.png)

### Control case

If I remove ``twinx`` it works:

```
import matplotlib.pyplot as plt

fig = plt.figure()
ax1 = fig.add_subplot(1,1,1)
ax2 = ax1.twinx()
ax1.plot([1,2,3])
fig.savefig('test.png')
```

![test](https://cloud.githubusercontent.com/assets/314716/6369593/2d15caee-bce8-11e4-9a73-ef209384d12e.png)

### Versions

```
$ python --version
Python 3.4.2 :: Continuum Analytics, Inc.
$ python -c 'import matplotlib; print(matplotlib.__version__)'
1.4.2
```

**Comment by tacaswell** [comment]
Duplicate of #3633 which should have been fixed by #3634 which should have been released with 1.4.3.  If this still exists in 1.4.3 please ping to have reopened.

**Comment by astrofrog** [comment]
@tacaswell - ah yes, it is indeed fixed now - sorry for the noise!
